crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -198,6 +198,24 @@ impl C8yEndPoint {
         // Hence just matching the host prefix.
         (host == c8y_http_host || host == c8y_mqtt_host).then_some(url)
     }
+
+    // Return the tenant url retrived from the local proxy url
+    //
+    // Return the url unchanged if not a proxy url.
+    pub fn c8y_url_from_proxy_url(&self, proxy_url: &str) -> Result<Url, InvalidUrl> {
+        let mut valid_url: Url = proxy_url.parse().map_err(|err| InvalidUrl {
+            url: proxy_url.to_string(),
+            err,
+        })?;
+
+        if self.proxy.is_proxy_url(&valid_url) {
+            valid_url.set_host(Some(&self.c8y_host)).unwrap();
+            valid_url.set_port(None).unwrap();
+            valid_url.set_scheme(Protocol::Https.as_str()).unwrap();
+        }
```

Comment:
That `/c8y` prefix added by the proxy also must be removed:
```suggestion
            valid_url.set_path(&valid_url.path().replace("/c8y", ""));
        }
```